artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPLargeMessageReader.java

```diff
@@ -77,39 +73,73 @@ public AMQPLargeMessageReader open() {
    }
 
    @Override
-   public Message readBytes(Delivery delivery) throws Exception {
+   public boolean readBytes(Delivery delivery) throws Exception {
       if (closed) {
          throw new IllegalStateException("AMQP Large Message Reader is closed and read cannot proceed");
       }
 
+      serverReceiver.connection.requireInHandler();
+
       final Receiver receiver = ((Receiver) delivery.getLink());
       final ReadableBuffer dataBuffer = receiver.recv();
 
+      final AMQPSessionCallback sessionSPI = serverReceiver.getSessionContext().getSessionSPI();
+
       if (currentMessage == null) {
-         final AMQPSessionCallback sessionSPI = serverReceiver.getSessionContext().getSessionSPI();
          final long id = sessionSPI.getStorageManager().generateID();
          currentMessage = new AMQPLargeMessage(id, delivery.getMessageFormat(), null,
                                                sessionSPI.getCoreMessageObjectPools(),
                                                sessionSPI.getStorageManager());
          currentMessage.parseHeader(dataBuffer);
-
+         logger.trace("Initializing current message {} on {}", currentMessage, this);
          sessionSPI.getStorageManager().largeMessageCreated(id, currentMessage);
+         sessionSPI.execute(() -> validateFile(currentMessage));
```

Comment:
This check seems to be mostly pointless now as the code doesn't use a continuation to resume delivery after validation it just throws it into another thread and then continues on as if it was successful

Reply:
It still needs to create the file. which should happen away from the Netty Thread. 

in case of an exception it will initialize the exception on the session... The current writes are still going through though but the exception would be initialized. I could interrupt the addBytes if there was an exception, but the possibility of having an exception there is actually so low that if any issues are happening with the storage, other things will happen to put the server down anyways through IOExceptions.

Reply:
I added a commit to be squashed addressing this